api/envoy/extensions/filters/network/tcp_proxy/v3/tcp_proxy.proto

```diff
@@ -67,8 +67,24 @@ message TcpProxy {
     option (udpa.annotations.versioning).previous_message_type =
         "envoy.config.filter.network.tcp_proxy.v2.TcpProxy.TunnelingConfig";
 
-    // The hostname to send in the synthesized CONNECT headers to the upstream proxy.
-    string hostname = 1 [(validate.rules).string = {min_len: 1}];
+    message AutoSni {
+      // Optional port used to create the request-target for all CONNECT request, i.e. CONNECT <hostname_from_sni>:<default_port>
```

Comment:
What does auto SNI have to do with CONNECT?

Reply:
The idea is to automatically set hostname in request-target for upstream connection based on SNI from downstream request. For more context please read [my comment](https://github.com/envoyproxy/envoy/issues/19612#issuecomment-1026727748) where I described in detail how it would work.

Reply:
Ack, got it.